lib/cdo/contact_rollups_validation.rb

```diff
@@ -269,6 +269,28 @@ class ContactRollupsValidation
       min: 0,
       max: 0
     },
+    {
+      # Double-check consistency of the complex (but necessary) query
+      # to mark contacts in the Teacher role
+      name: "Check that all contacts that match 'Teacher Query' "\
+            "are in Teacher role",
+      query: "SELECT COUNT(*) from contact_rollups_daily
+              WHERE (dashboard_user_id is not null
+              OR forms_submitted LIKE '%BringToSchool2013%'
+              OR forms_submitted LIKE '%ClassSubmission%'
+              OR forms_submitted LIKE '%DistrictPartnerSubmission%'
+              OR forms_submitted LIKE '%HelpUs2013%'
+              OR forms_submitted LIKE '%K5OnlineProfessionalDevelopmentPostSurvey%'
+              OR forms_submitted LIKE '%K5ProfessionalDevelopmentSurvey%'
+              OR forms_submitted LIKE '%ProfessionalDevelopmentWorkshop%'
+              OR forms_submitted LIKE '%ProfessionalDevelopmentWorkshopSignup%'
+              OR forms_submitted LIKE '%StudentNomination%'
+              OR forms_submitted LIKE '%TeacherNomination%'
```

Comment:
Good idea, I will mention that in a comment. Within reasonable bounds, I prefer to author test expectations separately. When the test pulls in expectations directly from the code, it's too easy for an unintentional code change to change the test as well and make it pass. I have learned this from the school of hard knocks! Obviously that's not the right strategy for every situation (if expectations are changing frequently, etc). But this is so static I don't mind duplication in the test.